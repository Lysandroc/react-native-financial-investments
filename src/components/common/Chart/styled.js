import styled from 'styled-components/native';
import Text from '../Text';

export const WrapperView = styled.View`
	flex: 1 100%;
	align-content: center;
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
`;

export const Title = styled(Text)`
	flex-basis: 100%;
	font-size: 18px;
	font-family: 'Poppins-MediumItalic';
`;
