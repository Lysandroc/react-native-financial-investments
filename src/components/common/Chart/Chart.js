import React from 'react';
import { WrapperView, Title } from './styled';

const Chart = ({ children, label }) => {
	return (
		<WrapperView>
			<Title dark center accessibilityLabel="period-label">
				Per√≠odo: {label}
			</Title>
			{children}
		</WrapperView>
	);
};

export default Chart;
