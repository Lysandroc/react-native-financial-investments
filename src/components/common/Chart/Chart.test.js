import React from 'react';
import { View, Text } from 'react-native';
import { render, cleanup } from '@testing-library/react-native';
import Chart from './Chart';

describe('should test <Chart/>', () => {
	let component;
	let props = {
		label: 'period text here',
		children: (
			<View>
				<Text accessibilityLabel="children-text">children here</Text>
			</View>
		),
	};

	beforeEach(() => {
		jest.clearAllMocks();
		component = render(<Chart {...props} />);
	});

	afterEach(cleanup);

	it('snapshot should be valid', () => {
		expect(component.baseElement).toMatchSnapshot();
	});

	it('should return earning value correctly', () => {
		const periodLabel = component.queryByLabelText('period-label');
		expect(periodLabel).toHaveTextContent(`PerÃ­odo: ${props.label}`);
	});

	it('should return temporary children value correctly', () => {
		const childrenText = component.queryByLabelText('children-text');
		expect(childrenText).toHaveTextContent('children here');
	});
});
