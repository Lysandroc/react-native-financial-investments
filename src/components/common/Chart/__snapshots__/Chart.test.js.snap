// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test <Chart/> snapshot should be valid 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "display": "flex",
            "flexBasis": "100%",
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
          },
        ]
      }
    >
      <Text
        accessibilityLabel="period-label"
        center={true}
        dark={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "color": "#1c2124",
              "flexBasis": "100%",
              "fontFamily": "Poppins-MediumItalic",
              "fontSize": 18,
              "fontWeight": "normal",
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
      >
        Per√≠odo: 
        period text here
      </Text>
      <View>
        <Text
          accessibilityLabel="children-text"
        >
          children here
        </Text>
      </View>
    </View>
  </View>
</View>
`;
